### PR TITLE
Enable SQUASHFS support in the default VM kernel

### PR DESCRIPTION
**Pull request by tuommaki**
Squashfs is useful for read-only disk images e.g. in case of large auxiliary trusted setup files.

### CODE REVIEW

File: src/builders/skopeo_builder.rs [hunk 1]
```diff
@@ -526,9 +526,129 @@ impl SkopeoSyslinuxBuilder {
...
+                    "CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE",
+                ],
+                false,
+            )
+            .context("Failed to set CONFIG_SQUASHFS_FRAGMENT_CACHE_SIZE value to kernel config")?;
+            Self::run_command(
+                &["scripts/config", "--disable", "CONFIG_SQUASHFS_EMBEDDED"],
+                false,
+            )
+            .context("Failed to disable CONFIG_SQUASHFS_EMBEDDED flag to kernel config ")?;
```
Comment (tuommaki) [inline src/builders/skopeo_builder.rs:649]: This is ugly as hell but does the job. We should probably come up with some neat system where we could easily describe feature sets and give them names. Like name of the feature set `enable_squashfs` and then some table where above enable/disable/set combinations are easily described.

**Comment by aleasims** [comment]
Are all trusted setups provided as filesystem images? Isn't it a very special case? What's wrong with providing it as archive?

**Comment by tuommaki** [comment]
> Are all trusted setups provided as SquashFS filesystem images? Isn't it a very special case? What's wrong with providing it as archive?

The thing with SquashFS is that you can build the disk image by single command invocation (`mksquashfs <dir> <disk-image-file>`). This is easy to document, easy to use and doesn't require `sudo` for example. I'd also imagine that one can do this easily on macOS (at least there seems to be [tools in homebrew](https://formulae.brew.sh/formula/squashfs))?

So with this, one can bundle any set of read-only files into VM as a separate input file. If files are submitted as an archive, I think it may be slightly less efficient (since you need to uncompress & write them through two(?) filesystems - the VM VirtFS and then the host one) whereas I think the SquashFS should be relatively well optimized for just mapping the file entries and uncompressing them directly into page cache.

TL;DR; This isn't only for trusted setups, but it will probably be the most common use case. I think SquashFS is simpler & faster compared to archive.

> If it's required only for this setup only, should we add this to builder?

What do you mean by "add this to builder"?

**Comment by aleasims** [comment]
> What do you mean by "add this to builder"?

I meant adding SquashFS to default kernel configuration in our VM builder.

Okay, let's enable it by default, I guess it won't hurt.

The implementation looks a bit ugly, I agree, but I suppose there is no easy way to simplify kernel configuration? If so, let's merge it.

**Comment by tuommaki** [comment]
> let's enable it by default, I guess it won't hurt.

Yeah, this should be totally fine :slightly_smiling_face: 

> The implementation looks a bit ugly, I agree, but I suppose there is no easy way to simplify kernel configuration? If so, let's merge it.

Not in general. I noticed that the build system starts asking questions if one only enables some of the flags, so in practice all of them needs to be set. Hence my thinking towards some kind of structure with feature name and then set of configs, but I think this is a good candidate for later refactoring, especially if we need to add other configs as well. Not a huge issue right now.

**Review by aleasims — APPROVED** [review]
